EuclideanGeometry/Example/SCHAUM/Problem1.1.lean

```diff
@@ -0,0 +1,79 @@
+import EuclideanGeometry.Foundation.Index
+
+noncomputable section
+
+namespace EuclidGeom
+
+variable {P : Type _} [EuclideanPlane P]
+
+namespace Problem_1_1
+/-Let $\triangle ABC$ be an isosceles triangle in which $AB = AC$.Let $D$ be a point on $AB$.
+Let $E$ be a point on $AC$ such that $AE = AD$. Let $M$ be the midpoint of $BC$.
+
+Prove that $DM = EM$.
+-/
+--Let $\triangle ABC$ be an isosceles triangle in which $AB = AC$.
+variable {A B C : P} {hnd: ¬ colinear A B C} {hisoc: (▵ A B C).IsIsoceles}
+--Let $D$ be a point on $AB$.
+variable {D : P} {D_on_seg: D LiesInt (SEG A B)}
+--Let $E$ be a point on $AC$
+variable {E : P} {E_on_seg: E LiesInt (SEG A C)}
+--such that $AE = AD$.
+variable {E_ray_position : (SEG A E).length = (SEG A D).length}
+--Let $M$ be the midpoint of $BC$.
+variable {M : P} {median_M_position : M = (SEG B C).midpoint}
+--Prove that $DM = EM$.
+theorem Problem_1_1 : (SEG D M).length = (SEG E M).length := by
+  have h₀ : (SEG A B).length = (SEG A C).length := by
+    calc
+      _ = (SEG C A).length := hisoc.symm
+      _ = (SEG A C).length := length_eq_length_of_rev (SEG C A)
+  have h₁ : ¬ colinear B D M := by sorry
+  have h₂ : ¬ colinear C E M := by sorry
+  --to confirm the definition of angle is not invalid
+  have d_ne_b : D ≠ B := (ne_of_not_colinear h₁).2.2
+  have m_ne_b : M ≠ B := (ne_of_not_colinear h₁).2.1.symm
+  have e_ne_c : E ≠ C := (ne_of_not_colinear h₂).2.2
+  have m_ne_c : M ≠ C := (ne_of_not_colinear h₂).2.1.symm
+  have a_ne_b : A ≠ B := (ne_of_not_colinear hnd).2.2.symm
+  have c_ne_b : C ≠ B := (ne_of_not_colinear hnd).1
+  have a_ne_c : A ≠ C := (ne_of_not_colinear hnd).2.1
+  have b_ne_c : B ≠ C := (ne_of_not_colinear hnd).1.symm
+  --the second edge of congruence
```

Comment:
the comment should be written as the style of proving in human-readable language, not comment of code.
`-- the length of $BD$ equals to length of $CE$.`